rebecca/stock-api-server/app/assets/javascripts/stock/stock.controller.js

```diff
@@ -0,0 +1,30 @@
+/* eslint no-param-reassign: 0, angular/controller-as: 0*/
+
+angular
+  .module('stockApp')
+  .controller('StockController', function($route, $location) {
+    var vm = this;
+
+    vm.sortType = 'rank';
+    vm.sortReverse = false;
+    vm.date = '';
+    vm.code = '';
+
+    vm.submitForm = function() {
+      $location.path('/search').search({ date: vm.date, code: vm.code });
+    };
+  })
+  .controller('DataController', function($scope, stocks) {
+    var vm = this;
```

Comment:
@joanna8030 
What is different with vm and $scope?
ref: http://stackoverflow.com/questions/16619740/angular-should-i-use-this-or-scope